vms/platformvm/state/state.go

```diff
@@ -352,6 +356,11 @@ type state struct {
 	subnetBaseDB  database.Database
 	subnetDB      linkeddb.LinkedDB
 
+	// Subnet ID --> Owner of the subnet
+	subnetOwners     map[ids.ID]fx.Owner
+	subnetOwnerCache cache.Cacher[ids.ID, *wrappedFxOwner] // cache of subnetID -> owner if the entry is nil, it is not in the database
```

Comment:
feel free to ignore this comment: It's consistent w/ the rest of the code but I personally don't think we need comments here.

Reply:
I agree, we'll keep it for now but I'll probably make a follow-up cleaning up the comments